mne/viz/evoked.py

```diff
@@ -1844,11 +1879,78 @@ def _get_ci_function_for_evokeds(ci):
     return _ci_fun
 
 
+def _finish_styles_plot_comp_evoked(styles, colors, linestyles, conditions,
+                                    cmap):
+    """Finalize styles for plot_compare_evokeds."""
+    # Styles (especially color and linestyle) are pulled from a dict 'styles'.
+    # This dict has one entry per condition. Its color and linestyle entries
+    # are pulled from the 'colors' and 'linestyles' dicts via '/'-tag matching
+    # unless they are overwritten by entries from a user-provided 'styles'.
+
+    # first, copy to avoid overwriting
+    styles = deepcopy(styles)
+    colors = deepcopy(colors)
+    linestyles = deepcopy(linestyles)
```

Comment:
you don't see to overwrite in this function. Can you do the copies just before the overwrites so it's clearer why you need it?